hack/machineset.sh

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env bash
+
+# machineset.sh - create a Windows MachineSet for Windows Machine Config Operator
+# Creates a MachineSet.yaml file and apply/delete the MachineSet if optional action is provided
+#
+# USAGE
+#    machineset.sh
+# OPTIONS
+#    $1      Action       (Optional) apply/delete the MachineSet
+# PREREQUISITES
+#    oc                   to fetch cluster info and apply/delete MachineSets on the cluster(cluster should be logged in)
+#    aws                  to fetch Windows AMI id for AWS platform (only required for clusters running on AWS)
+
+function error-exit() {
+    echo "Error: $*" >&2
+    exit 1
+}
+
+ACTION=$1
+
+# get_spec returns the template yaml common for all cloud providers
+get_spec(){
+
+local infraID=$1
+local az=$2
+local provider=$3
+
+machineSetName="$infraID"-windows-worker-"$az"
+if [ "$provider" = "azure" ]; then
+  machineSetName="winworker"
+fi
+
+cat <<EOF
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+  name: $(echo $machineSetName)
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+      machine.openshift.io/cluster-api-machineset: $(echo $machineSetName)
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: $(echo $machineSetName)
+        machine.openshift.io/os-id: Windows
+    spec:
+      metadata:
+        labels:
+          node-role.kubernetes.io/worker: ""
+EOF
+}
+
+# get_aws_ms creates a MachineSet for AWS Cloud Provider
+get_aws_ms(){
+
+local infraID=$1
+local region=$2
+local az=$3
+local provider=$4
+
+# get the AMI id for the Windows VM
+ami_date="2020.09.09"
+ami_id=$(aws ec2 describe-images --filters Name=name,Values=Windows_Server-1909-English-Core-ContainersLatest-${ami_date} --region ${region} | awk '{print $17}' | head -n1)
```

Comment:
The bug that I [found](https://github.com/openshift/windows-machine-config-operator/pull/121#issuecomment-704480941) is with `print $17`. I suggest using `jq` instead to parse the output.

Reply:
Fixed